### PR TITLE
Implement support for compressed instructions.

### PR DESCRIPTION
**Pull request by user-0xcafe**
This PR should be only merged after the PR in `cksystemsgroup/riscu` is merged. 
It implements the new interface for iterating over variable length instructions 
in the _disassembler_ and the _emulator_.

### CODE REVIEW

File: src/emulate.rs [hunk 1]
```diff
@@ -126,37 +128,30 @@ impl EmulatorState {
...
+        LittleEndian::read_u64(&self.memory[adr as usize..])
     }
 
     fn get_mem_typed<T: MyLittleEndian>(&self, adr: EmulatorValue) -> T {
-        assert!(adr % (size_of::<T>() as u64) == 0, "address aligned");
-        let word_address = adr & !WORD_SIZE_MASK;
-        let word_offset = (adr & WORD_SIZE_MASK) as usize;
-        let bytes = self.get_mem(word_address).to_le_bytes();
-        MyLittleEndian::read(&bytes[word_offset..])
+        MyLittleEndian::read(&self.memory[adr as usize..])
```
Comment (mstarzinger) [inline src/emulate.rs:137]: nitpick: Let's bring back the alignment check in the `get_mem_typed` method. This way our emulator is strict by default and only allows unaligned access when the explicit `get_mem_unaligned` below is used.

File: src/emulate.rs [hunk 2]
```diff
@@ -335,15 +329,15 @@ fn exec_jal(state: &mut EmulatorState, jtype: JType) {
 // pc = rs1 + s64(imm)
 fn exec_jalr(state: &mut EmulatorState, itype: IType) {
     let rs1_value = state.get_reg(itype.rs1());
-    let rd_value = state.program_counter + riscu::INSTRUCTION_SIZE as u64;
+    let rd_value = state.program_counter + (state.current_instruction_lenght() as u64);
```
Comment (mstarzinger) [inline src/emulate.rs:334]: nitpick: We should also adapt the comment above this method for consistency.

**Review by mstarzinger — APPROVED** [review]
Nice! Looks good to me. The changes to `riscu` have been merged, we should now be able to bump the dependency in Cargo.lock as part of this PR as well using the `$ cargo update -p riscu` command.

**Comment by user-0xcafe** [comment]
Sorry for force-pushing. I accidentally triggered a rebase between unicorn and my working copy.

File: src/emulate.rs (hunk 1 above)
Comment (user-0xcafe) [reply src/emulate.rs:137]: Taking it out was an oversight on my side. Fixed it.

File: src/emulate.rs (hunk 2 above)
Comment (user-0xcafe) [reply src/emulate.rs:334]: Done.

**Pull request merged**